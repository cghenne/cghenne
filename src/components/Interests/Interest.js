// @flow
import React from 'react';
import './Interests.css';

type Props = {
  label: string,
  description: string
};

const Interest = ({ label, description }: Props) => {
  return (
    <li>
      <span className="bold">{label}</span>: {description}
    </li>
  );
};

Interest.displayName = 'Interest';

export default Interest;
