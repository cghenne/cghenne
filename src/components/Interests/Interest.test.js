import React from 'react';
import { shallow } from 'enzyme';
import Interest from './Interest';

const fakeInterest = {
  label: 'Board Games',
  description: 'Monopoly, Settlers of Catans, Cards Against Humanity'
};

describe('Interest', () => {
  const wrapper = shallow(<Interest {...fakeInterest} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the interest in a li tag', () => {
    expect(wrapper.find('li').exists()).toEqual(true);
  });

  it('should render the interest label in bold', () => {
    expect(wrapper.find('.bold').text()).toEqual(fakeInterest.label);
  });

  it('should render the interest description', () => {
    expect(wrapper.find('li').text()).toContain(fakeInterest.description);
  });
});
