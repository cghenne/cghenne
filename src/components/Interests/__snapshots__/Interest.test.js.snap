// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interest should render without crashing 1`] = `
<li>
  <span
    className="bold"
  >
    Board Games
  </span>
  : 
  Monopoly, Settlers of Catans, Cards Against Humanity
</li>
`;
